rust/lance/src/index/cache.rs

```diff
@@ -35,6 +36,7 @@ pub struct IndexCache {
     // TODO: Can we merge these two caches into one for uniform memory management?
     scalar_cache: Arc<Cache<String, Arc<dyn ScalarIndex>>>,
     vector_cache: Arc<Cache<String, Arc<dyn VectorIndex>>>,
+    frag_reuse_cache: Arc<Cache<String, Arc<FragReuseIndex>>>,
```

Comment:
🤣 😭  I will consolidate these soon...

Reply:
yeah I saw the TODO so was thinking it could be done together, currently I am just keeping things consistent...